Puzanau Andrei

apuzanov opened the pull request:
FYI - 17.12.2019
Добавил "вольную" реализацию итерацию 6 для того, чтобы при возможности обсудить используемый подход.

apuzanov committed:
Add an initial version of the RSS-reader based on first iteration requirements

apuzanov committed:
Utility is wrapped into distribution package

apuzanov committed:
Utility is wrapped into distribution package

apuzanov committed:
Merge branch 'final_task' of https://github.com/apuzanov/PythonHomework into final_task

AlexeiBuzuma commented on rssreader/feed.py:
```diff
@@ -0,0 +1,75 @@
+
+class FeedEncoder(json.JSONEncoder):
+    """Subclass of JSONEncoder to be used for an instance of Feed class transforming into JSON"""
+    def default(self, obj) -> Dict:
```
С моей точки зрения тайп хинты должны либо нигде не использоваться, либо использоваться везде. 
Советую их везде прописать (имею в виду аргументы функций и возвращаемые значения) :)

AlexeiBuzuma commented on tests/__init__.py:
```diff
@@ -0,0 +1,3 @@
+"""
+Tests are implemented using standard unittest framework. Unit tests provide tests for separate classes/modules.
+"""
```
`pycodestyle` дает пару стилистических ошибок
```bash
$ pycodestyle --max-line-length=120 .
./tests/__init__.py:3:4: W292 no newline at end of file
./tests/test_feed.py:145:1: W391 blank line at end of file
```

AlexeiBuzuma commented on rssreader/feed.py:
```diff
@@ -0,0 +1,75 @@
+import html
+from typing import Dict, List
+import json
+import feedparser
```
Нужно отсортировать все импорты согласно PEP8

https://www.python.org/dev/peps/pep-0008/#imports

AlexeiBuzuma commented on setup.py:
```diff
@@ -0,0 +1,22 @@
+    author_email="puzanov.a.a@gmail.com",
+    packages=[conf.__package__],
+    install_requires=[
+        "feedparser==5.2.1",
```
Советую вынести зависимости в отдельный файл `requirements.txt` 
Так этими зависимостями смогут пользоваться люди, которые не планируют устанавливать твой пакет + список зависимостей имеет свойство разрастаться, в отдельном файле ими управлять проще.

apuzanov replied:
Сделал следующим образом: requirements.txt вношу в MANIFEST.in (чтобы попал в дистрибутив) и загружаю содержимое в install_requires в setup.py через pip._internal.req.parse_requirements.
Верный путь?

apuzanov replied:
Итог: все зависимости в requirements.txt, setup.py про них ничего не знает.

apuzanov committed:
RSS news is stored in a local storage while reading

All RSS news are stored in a local cache (based on sqlite3) when data are downloaded from provided url. Cache contains unique set of news - duplicates are skipped while saving.
The cached news are read when --date parameter is specified. In case of no news an error occures.
Additional changes\fixes:
 - all required packages are in standalone file requirements.txt
 - all imports are sorted according to PEP8
 - type hints are added everwhere

apuzanov committed:
News can be converted to HTML and FB2 formats

Two new arguments to-html and to-fb2 are added to convert news to HTML and FIctionBook2 formats respectively.
Value of these arguments is a new file path to store convertion result. File with such name should not exist. Otherwise, an error occurs.
If image is skipped in news, default one is used. As fb2 supports on jpeg and png formats, default image is used instead of all other formats.
There are also several changes\fixes:
 - setup.py contains required additonal packages as requirements.txt
 - cached news\images are stored in "home"/.rssreader folder

apuzanov committed:
Add opportunity to print the result of the utility in colorized mode

apuzanov committed:
Tests are divided into unit and integration

Existent tests are devided into unit and integration.
Additionally several issues are fixed:
- if data are goting to be loaded from local cache but there is no cache at this time, DB is not currenctly created
- during converting application limit argument is taken into account

AlexeiBuzuma commented on rssreader/converter.py:
```diff
@@ -0,0 +1,263 @@
+        super().__init__(cache_dir, file_path)
+
+        self.book_template = '''\
+<?xml version="1.0" encoding="{encoding}"?>
```
Это не критично, но можно подумать над тем, чтобы хранить такого рода шаблоны в отдельных файлах (например как джанго темплейты).

apuzanov committed:
Web-server implementation based on Flask-RESTful.
